site/src/hooks/useClickable.ts

```diff
@@ -1,20 +1,45 @@
-import { KeyboardEvent } from "react";
+import {
+  type MouseEventHandler,
+  type KeyboardEvent,
+  type RefObject,
+  useRef,
+} from "react";
 
-export interface UseClickableResult {
+// Literally any object (ideally an HTMLElement) that has a .click method
+type ClickableElement = {
+  click: () => void;
+};
+
+export interface UseClickableResult<
+  T extends ClickableElement = ClickableElement,
+> {
+  ref: RefObject<T>;
   tabIndex: 0;
   role: "button";
-  onClick: () => void;
+  onClick: MouseEventHandler<T>;
   onKeyDown: (event: KeyboardEvent) => void;
 }
 
-export const useClickable = (onClick: () => void): UseClickableResult => {
+export const useClickable = <
+  // T doesn't have a default type to make it more obvious that the hook expects
+  // a type argument in order to work at all
+  T extends ClickableElement,
+>(
+  onClick: MouseEventHandler<T>,
+): UseClickableResult<T> => {
+  const ref = useRef<T>(null);
+
   return {
+    ref,
     tabIndex: 0,
     role: "button",
     onClick,
     onKeyDown: (event: KeyboardEvent) => {
```

Comment:
Something I just realized: even though the event handler is called `onClick`, I'm pretty sure that it supports the user clicking Space+Enter right out of the box, with no extra keyboard handlers needed